exporters/committime/collector_base.py

```diff
@@ -92,42 +92,63 @@ def generate_metrics(self) -> Iterable[CommitMetric]:
         for namespace in self._namespaces:
             # Initialized variables
             builds = []
+            pipeline_runs = []
             apps = []
+            pipelines = []
             builds_by_app = {}
+            runs_by_app = {}
             app_label = pelorus.get_app_label()
             logging.debug(
                 "Searching for builds with label: %s in namespace: %s"
                 % (app_label, namespace)
             )
-
+            # This get all Builds in our cluster
             v1_builds = self._kube_client.resources.get(
                 api_version="build.openshift.io/v1", kind="Build"
             )
+            v1_tekton = self._kube_client.resources.get(
+                api_version="tekton.dev/v1beta1", kind="PipelineRun"
+            )
             # only use builds that have the app label
             builds = v1_builds.get(namespace=namespace, label_selector=app_label)
-
+            # only use PipelineRun that have the app label
+            pipeline_runs = v1_tekton.get(namespace=namespace, label_selector=app_label)
             # use a jsonpath expression to find all values for the app label
             jsonpath_str = (
                 "$['items'][*]['metadata']['labels']['" + str(app_label) + "']"
             )
             jsonpath_expr = parse(jsonpath_str)
 
             found = jsonpath_expr.find(builds)
+            runs_found = jsonpath_expr.find(pipeline_runs)
 
             apps = [match.value for match in found]
+            pipelines = [match.value for match in runs_found]
 
-            if not apps:
+            if not apps and not pipelines:
                 continue
-            # remove duplicates
-            apps = list(dict.fromkeys(apps))
-            builds_by_app = {}
-
-            for app in apps:
-                builds_by_app[app] = list(
-                    filter(lambda b: b.metadata.labels[app_label] == app, builds.items)
-                )
-
-            metrics += self.get_metrics_from_apps(builds_by_app, namespace)
+            elif apps:
```

Comment:
@jlozano-rh this means that only apps _or_ pipelines will be checked, not both. Is that intentional?

Reply:
@KevinMGranger no, my bad... I made a mistake there, the idea is the following: if there are no apps or pipelines to process it does the `continue` statement and avoid processing something empty. And then the idea is to get metrics from what is actually there, I mean, if there are no pipelines to process... there is no need to get metrics from them because there are none. You know what I mean?

Reply:
That makes sense!

Our linter was complaining about cyclomatic complexity. Since both `get_metrics_from_pipelineruns` and `get_metrics_from_apps` loop over what's given to them, passing them something empty won't be an issue. That means we can remove another level of nesting.

Reply:
@KevinMGranger yes, it would even look better... my code is very primitive given the urgency I had at the time with the customer.

Reply:
If it works, it works! Just wanted to make sure I wasn't about to make the mistake of [Chesterton's Fence](https://fs.blog/chestertons-fence/).